index.js

```diff
@@ -87,39 +87,48 @@ function getProjectManifest(projectDirName) {
   };
 }
 
-exports.main = function main() {
-  if (!process.argv[2]) {
-    console.error(`${chalk.red("Missing project dir name.")}\n`);
-    console.log(USAGE_MSG);
-    process.exit(1);
+function main({
+  dirPath,
+  baseDir = process.cwd(),
+  getProjectManifestFn = getProjectManifest,
```

Comment:
Do we need all this options here?
I think we should evaluate these options and decide if it makes sense for an app that includes this module as a library to redefine this helpers or if we need them only for testing reasons.

If they are only needed for testing reasons, with jest we can probably just move them in another module file and use the Jest Mocks feature to mock them (http://facebook.github.io/jest/docs/manual-mocks.html)

Reply:
@rpl i used auto mocking in this case, I think it acheives what you mean